client/my-sites/stats/mobile-promo-cards/index.jsx

```diff
@@ -0,0 +1,82 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { MobilePromoCard } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import wordpressSeoIllustration from 'calypso/assets/images/illustrations/wordpress-seo-premium.svg';
+import PromoCardBlock from 'calypso/blocks/promo-card-block';
+import DotPager from 'calypso/components/dot-pager';
+
+import './style.scss';
+
+const EVENT_YOAST_PROMO_VIEW = 'calypso_stats_wordpress_seo_premium_banner_view';
+const EVENT_MOBILE_PROMO_VIEW = 'calypso_stats_traffic_mobile_cta_jetpack_view';
+
+export default function MobilePromoCardWrapper( { isJetpack, isOdysseyStats, slug } ) {
+	// Yoast promo is disabled for Odyssey & self-hosted.
+	// Mobile apps promos are always shown.
+	const showYoastPromo = ! isOdysseyStats && ! isJetpack;
+	const showBothPromoCards = showYoastPromo;
+	// Handle initial view event if not using the DotPager UI.
+	// We don't worry about the Yoast card as it sends on mount.
+	useEffect( () => {
+		if ( ! showBothPromoCards ) {
+			recordTracksEvent( EVENT_MOBILE_PROMO_VIEW );
+		}
+	}, [ showBothPromoCards ] );
+	// Handle click events from promo card.
+	const promoCardDidReceiveClick = ( event ) => {
+		// Events need to incorporate the page and the click type.
+		// This will allow us to account for integraton across different pages.
+		// FORMAT: calypso_stats_PAGE_mobile_cta_EVENT
+		const tracksEventName = `calypso_stats_traffic_mobile_cta_${ event.replaceAll( '-', '_' ) }`;
+		recordTracksEvent( tracksEventName );
+	};
+	// Handle view events from DotPager UI.
+	const pagerDidSelectPage = ( index ) => {
+		const evenLookup = [ EVENT_YOAST_PROMO_VIEW, EVENT_MOBILE_PROMO_VIEW ];
+		const eventName = evenLookup[ index ];
+		recordTracksEvent( eventName );
+	};
+	// Render one or both promo cards.
+	return (
+		<>
+			{ ! showBothPromoCards && (
+				<div className="stats__promo-container">
+					<div className="stats__promo-card">
+						<MobilePromoCard
+							className="stats__promo-card-apps"
+							clickHandler={ promoCardDidReceiveClick }
+						/>
+					</div>
+				</div>
+			) }
+			{ showBothPromoCards && (
+				<div className="stats__promo-container">
+					<div className="stats__promo-card">
+						<DotPager className="stats__promo-pager" onPageSelected={ pagerDidSelectPage }>
```

Comment:
This probably requires a non-trivial change to the DotPager component, but it'd be nice if we could fire an `EVENT_MOBILE_PROMO_VIEW` event when the user pages to the `MobilePromoCard`.

Reply:
I'm not sure I follow here @jsnmoon.

The current implementation does fire a VIEW event any time the user pages to a new card. It also fires a VIEW event on initial page load for whichever card is the default.

So visiting Stats with the DotPager component active then paging back and forth would fire the following:

1. EVENT_YOAST_PROMO_VIEW — on page load
2. EVENT_MOBILE_PROMO_VIEW — on card change to mobile promo
3. EVENT_YOAST_PROMO_VIEW — no card change back to Yoast promo

If there is a case I'm missing here, please let me know!

Reply:
Aha! I was mistaken -- I didn't know that we would fire the `EVENT_MOBILE_PROMO_VIEW` event when paging through the DotPager component. Please disregard!